frame/staking/src/pallet/mod.rs

```diff
@@ -551,6 +604,80 @@ pub mod pallet {
 	#[pallet::getter(fn current_planned_session)]
 	pub type CurrentPlannedSession<T> = StorageValue<_, SessionIndex, ValueQuery>;
 
+	/// Wrapper struct for `ErasRewards`, `ClaimedRewards`, `ErasRewardsClipped`,
+	/// `ErasRewardsPaged`, `ErasTotalStake`.
+	pub(crate) struct EraInfo<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> EraInfo<T> {
+		// TODO(ank4n): doc and clean up in 84 eras
+		// looks at ledger for older non paged rewards, and `ClaimedRewards` for newer paged
+		// rewards.
+		pub(crate) fn temp_is_rewards_claimed(
```

Comment:
This is a bit ugly function that will check both the old ledger claimed rewards and the new paged claimed rewards. I can keep it outside `EraInfo` but then I do not want to expose `is_rewards_claimed ` publicly since it will just add to confusion which one is the right one. I feel this is the way to go till we can clean this up.